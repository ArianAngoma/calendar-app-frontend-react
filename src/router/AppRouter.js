import {
    BrowserRouter as Router,
    Switch,
    // Route,
    Redirect
} from 'react-router-dom';
import {useDispatch, useSelector} from 'react-redux';

/* Importaciones propias */
import {LoginScreen} from "../components/auth/LoginScreen";
import {CalendarScreen} from "../components/calendar/CalendarScreen";
import {useEffect} from 'react';
import {startChecking} from '../actions/auth';
import {PublicRoute} from './PublicRoute';
import {PrivateRoute} from './PrivateRoute';

export const AppRouter = () => {
    const dispatch = useDispatch();

    /* Estado de la autenticaciÃ³n */
    const {checking, uid} = useSelector(state => state.auth);

    /* Validar si tengo un usuario logueado */
    useEffect(() => {
        dispatch(startChecking());
    }, [dispatch]);

    if (checking) {
        return (
            <div className="m-0 vh-100 row justify-content-center align-items-center">
                <div className="spinner-border text-primary"></div>
            </div>
        );
    }

    return (
        <Router>
            <div>
                <Switch>
                    {/*<Route exact path="/login" component={LoginScreen}/>
                    <Route exact path="/" component={CalendarScreen}/>*/}

                    <PublicRoute exact path="/login" component={LoginScreen} isAuthenticated={!!uid}/>
                    <PrivateRoute exact path="/" component={CalendarScreen} isAuthenticated={!!uid}/>

                    <Redirect to="/"/>
                </Switch>
            </div>
        </Router>
    )
}