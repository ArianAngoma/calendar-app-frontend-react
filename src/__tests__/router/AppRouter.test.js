import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

/* Importaciones propias */
import {AppRouter} from '../../router/AppRouter';

/* Configuración del Store */
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Pruebas en <AppRouter/>', () => {
    test('Debería de mostrar el Loading', () => {
        /* Estado inicial del Store */
        const initState = {
            auth: {
                checking: true
            }
        };
        let store = mockStore(initState);

        const wrapper = mount(
            <Provider store={store}>
                <AppRouter/>
            </Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });
});