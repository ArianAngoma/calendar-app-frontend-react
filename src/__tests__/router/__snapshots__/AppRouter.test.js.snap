// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> Deber√≠a de mostrar el Loading 1`] = `
<div
  className="m-0 vh-100 row justify-content-center align-items-center"
>
  <div
    className="spinner-border text-primary"
  />
</div>
`;
